src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -704,6 +704,15 @@ void Unit::DealDamageMods(Unit const* victim, uint32 &damage, uint32* absorb) co
 
 uint32 Unit::DealDamage(Unit* victim, uint32 damage, CleanDamage const* cleanDamage, DamageEffectType damagetype, SpellSchoolMask damageSchoolMask, SpellInfo const* spellProto, bool durabilityLoss)
 {
+    // Sparring Checks
+    if (Creature* me = ToCreature())
+        if (Creature* target = victim->ToCreature())
+            if (me->CanSparrWith(target))
```

Comment:
If you put else, you need add braces here

Reply:
OK, so "around the lines 711-714" like I mentioned in my comment above, then.